services/ui-src/src/utils/auth/userProvider.tsx

```diff
@@ -35,8 +34,27 @@ export const UserProvider = ({ children }: Props) => {
 
   const checkAuthState = useCallback(async () => {
     try {
-      const authenticatedUser = await Auth.currentAuthenticatedUser();
-      setUser(authenticatedUser);
+      const session = await Auth.currentSession();
+      const { email, given_name, family_name } = await session.getIdToken()
+        .payload;
+      // "custom:cms_roles" is an string of concat roles so we need to check for the one applicable to MCR
+      const userRole = (
+        (await session.getIdToken().payload?.["custom:cms_roles"]) as
+          | string
+          | undefined
```

Comment:
Since we are calling pulling the payload off multiple times, it may help to store it in it's own local variable for readability.

Something like this maybe

```
const payload = session.getIdToken().payload;
const { email, given_name, family_name } = payload;
....
const cms_role = payload["custom:cms_roles"] as string ?? "";
const userRole = cms_role.split(",").find((r) => r.includes("mdctmcr"));
...
```